hudi-client/hudi-client-common/src/main/java/org/apache/hudi/index/bucket/ConsistentBucketIndexUtils.java

```diff
@@ -117,55 +116,74 @@ public static Option<HoodieConsistentHashingMetadata> loadMetadata(HoodieTable t
         return filename.contains(HASHING_METADATA_FILE_SUFFIX);
       };
       final List<StoragePathInfo> metaFiles = metaClient.getStorage().listDirectEntries(metadataPath);
-      final TreeSet<String> commitMetaTss = metaFiles.stream().filter(hashingMetaCommitFilePredicate)
-          .map(commitFile -> HoodieConsistentHashingMetadata.getTimestampFromFile(commitFile.getPath().getName()))
-          .sorted()
-          .collect(Collectors.toCollection(TreeSet::new));
-      final List<StoragePathInfo> hashingMetaFiles = metaFiles.stream().filter(hashingMetadataFilePredicate)
-          .sorted(Comparator.comparing(f -> f.getPath().getName()))
+
+      final TreeMap<String/*instantTime*/, Pair<StoragePathInfo/*hash metadata file path*/, Boolean/*commited*/>> versionedHashMetadataFiles = metaFiles.stream()
+          .filter(hashingMetadataFilePredicate)
+          .map(metaFile -> {
+            String instantTime = HoodieConsistentHashingMetadata.getTimestampFromFile(metaFile.getPath().getName());
+            return Pair.of(instantTime, Pair.of(metaFile, false));
+          })
+          .sorted(Collections.reverseOrder())
+          .collect(Collectors.toMap(Pair::getLeft, Pair::getRight, (a, b) -> a, TreeMap::new));
+
+      metaFiles.stream().filter(hashingMetaCommitFilePredicate)
+          .forEach(commitFile -> {
+            String instantTime = HoodieConsistentHashingMetadata.getTimestampFromFile(commitFile.getPath().getName());
+            if (!versionedHashMetadataFiles.containsKey(instantTime)) {
```

Comment:
Can we just fix the metadata path and keep the other logic as it is so that it is more easier to review, if you wanna a refactoring to the code, let's fire another PR to address it.

Reply:
> Can we just fix the metadata path and keep the other logic as it is so that it is more easier to review, if you wanna a refactoring to the code, let's fire another PR to address it.

I originally intended to do this like u said, but found that fixing the metadata path code exposed the logic of the original code, assuming that there is now an inflight clustering, when loading metadata, according to the original code, it will return empty, which is not expected. Therefore, I fixed these two problems at the same time, and it was difficult to fix them separately, because the original unit-test would be failed.

Reply:
Let's try to fix the issues one by one

Reply:
> Let's try to fix the issues one by one

Hi! I have updated this pr. This pr will only fix the problem with the wrong file path, and I will initiate a new pr after this pr is merged with the changes in refactoring metadata management.

Reply:
> Let's try to fix the issues one by one

As you said, after we merged wrong path problem fixing pr, I raise a new pr to refactor consistent-hash-metadata management for better code readability and maintainability, lets keep reviewing at: https://github.com/apache/hudi/pull/12512